### ISSUE TITLE
New check: JavadocNoErrorInThrowsTag

### ISSUE THREAD

**Issue by gianmarcoschifone** [checkstyle/checkstyle#20792]
# Description
After https://github.com/checkstyle/checkstyle/issues/19923#issuecomment-4986750659

Documentation Comments Style says that `Error` types should not be documented because they are unpredictable.

Currently there is no dedicated Checkstyle check ensuring that Javadoc `@throws` and `@exception` tags do not document `Error` types. As a result, Javadoc comments may document serious JVM-level problems such as `OutOfMemoryError`, `StackOverflowError`, `AssertionError`, `LinkageError`, or `java.lang.Error`.

This check should report Javadoc `@throws` and `@exception` tags whose documented exception type is an `Error` type.

The check should report names such as:

- `Error`
- `java.lang.Error`
- `OutOfMemoryError`
- `StackOverflowError`
- `AssertionError`
- `LinkageError`
- fully qualified names whose simple name ends with `Error`

# Config

```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocNoErrorInThrowsTag"/>
    </module>
</module>
```

# Example

```java
import java.io.IOException;

public class Test {

    /**
     * Valid Javadoc.
     *
     * @throws IOException if an input or output error occurs
     * @throws IllegalArgumentException if the argument is invalid
     */
    public void valid1() throws IOException { }

    /**
     * Invalid Javadoc.
     *
     * @throws Error if a serious problem occurs // violation
     */
    public void invalid1() { }

    /**
     * Invalid Javadoc.
     *
     * @throws OutOfMemoryError if memory is exhausted // violation
     */
    public void invalid2() { }

    /**
     * Invalid Javadoc.
     *
     * @exception StackOverflowError if recursion is too deep // violation
     */
    public void invalid3() { }

    /**
     * Invalid Javadoc.
     *
     * @throws java.lang.AssertionError if an assertion fails // violation
     */
    public void invalid4() { }
}
```

# Output

```
Starting audit...
[ERROR] Test.java:/:/: Error type 'Error' should not be documented in @throws tag. [JavadocNoErrorInThrowsTag]
[ERROR] Test.java:/:/: Error type 'OutOfMemoryError' should not be documented in @throws tag. [JavadocNoErrorInThrowsTag]
[ERROR] Test.java:/:/: Error type 'StackOverflowError' should not be documented in @exception tag. [JavadocNoErrorInThrowsTag]
[ERROR] Test.java:/:/: Error type 'java.lang.AssertionError' should not be documented in @throws tag. [JavadocNoErrorInThrowsTag]
Audit done.
Checkstyle ends with 4 errors.
```

**Comment by sahitya0xsingh** [comment]
I'd like to work on this issue.

**Comment by romani** [comment]
@sahitya0xsingh , we recommend to try bug fix before creation of new Check, it will help you undrstand how checkstyle works before you experiencing top complexity.

**Comment by sahitya0xsingh** [comment]
Thank you for the recommendation! I'll start by working on a bug fix to become more familiar with the Checkstyle codebase, and then I'll come back to this issue.

**Comment by romani** [comment]
Please add Error name to message to let users suppress some violations if they named exception as xxxxError.

**Comment by romani** [comment]
https://checkstyle.org/styleguides/doc-comments-style/doc-comments-styleguide.html#throws-tag

>Errors should not be documented as they are unpredictable.

In our case they are predictable, so it not a case for violation.

Please update doc of Check that we will not violate Errors that are explicitly  called in code.

**Comment by romani** [comment]
@gianmarcoschifone , please fix link in description after word "After"
